fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/ResolvedKeySpacePathTest.java

```diff
@@ -0,0 +1,232 @@
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+/**
+ * Tests for {@link ResolvedKeySpacePath}.
+ */
+@Tag(Tags.RequiresFDB)
+class ResolvedKeySpacePathTest {
+    @RegisterExtension
+    final FDBDatabaseExtension dbExtension = new FDBDatabaseExtension();
+
+    /**
+     * Test value pairs for each KeyType.
+     */
+    private static final Map<KeyType, TestValuePair> TYPE_TEST_VALUES = Map.of(
+            KeyType.STRING, new TestValuePair(() -> "value1", () -> "value2"),
+            KeyType.LONG, new TestValuePair(() -> 100L, () -> 200L),
+            KeyType.BYTES, new TestValuePair(() -> new byte[]{1, 2, 3}, () -> new byte[]{4, 5, 6}),
+            KeyType.UUID, new TestValuePair(() -> new UUID(1, 1), () -> new UUID(2, 2)),
+            KeyType.BOOLEAN, new TestValuePair(() -> true, () -> false),
+            KeyType.NULL, new TestValuePair(() -> null, () -> null),
+            KeyType.FLOAT, new TestValuePair(() -> 1.5f, () -> 2.5f),
+            KeyType.DOUBLE, new TestValuePair(() -> 1.5d, () -> 2.5d)
+    );
+
+    @Nonnull
+    static Stream<Arguments> testEqualsHashCode() {
+        return ParameterizedTestUtils.cartesianProduct(
+                Arrays.stream(KeyType.values()),
+                ParameterizedTestUtils.booleans("constantDirectory"),
+                ParameterizedTestUtils.booleans("differenceInParent")
+        );
+    }
+
+    /**
+     * Test equals and hashCode contracts for depth 1 directories.
+     */
+    @ParameterizedTest
+    @MethodSource("testEqualsHashCode")
+    void testEqualsHashCode(@Nonnull KeyType keyType, boolean constantDirectory, boolean differenceInParent) {
+        @Nonnull TestValuePair values = TYPE_TEST_VALUES.get(keyType);
+        // Test case 1: Same logical and resolved values (existing test)
+        ResolvedKeySpacePath path1 = createResolvedPath(keyType, values.getValue1(), values.getValue1(),
+                createRootParent(), constantDirectory, differenceInParent);
+        ResolvedKeySpacePath path2 = createResolvedPath(keyType, values.getValue1(), values.getValue1(),
+                createRootParent(), constantDirectory, differenceInParent);
+
+        // Test equality contracts
+        assertEquals(path1, path2, "Identical paths should be equal");
+        assertEquals(path2, path1, "Symmetry: path2.equals(path1)");
+        assertEquals(path1.hashCode(), path2.hashCode(), "Equal objects must have equal hash codes");
+            
+        // Test inequality when values differ (except NULL type which only has null values)
+        if (keyType != KeyType.NULL && values.getValue2() != null) {
```

Comment:
Isn't this redundant (if `keyType == KeyType.NULL` then both value1 and value2 are null)?